### PR TITLE
User friendlier signup errors

### PR DESCRIPTION
**Pull request by crufter**

### CODE REVIEW

File: signup/handler/signup.go [hunk 1]
```diff
@@ -152,14 +154,15 @@ func (e *Signup) sendVerificationEmail(ctx context.Context,
 
 	_, isAllowed := e.isAllowedToSignup(ctx, req.Email)
 	if !isAllowed {
-		return merrors.Forbidden("signup.notallowed", "user has not been invited to sign up")
+		return merrors.Forbidden("signup.notallowed", "user %v has not been invited to sign up", req.Email)
```
Comment (domwong) [inline signup/handler/signup.go:157]: We should probably change the error to something a bit nicer like "We're currently operating an invite only beta and this email address has not been invited"

Comment (crufter) [reply signup/handler/signup.go:157]: I love that! :+1:

File: signup/handler/signup.go [hunk 2]
```diff
@@ -31,6 +31,11 @@ import (
 	mstore "github.com/micro/micro/v3/service/store"
 )
 
+const (
+	internalErrorMsg  = "Something went wrong during signup. Please try again a bit later. If the issue persists, let us know on Slack"
+	notInvitedErroMsg = "We're currently operating an invite only beta and this email address has not been invited"
```
Comment (ben-toogood) [inline signup/handler/signup.go:36]: Typo Erro => Error

File: signup/handler/signup.go (hunk 1 above)
Comment (asim) [reply signup/handler/signup.go]: Do not mention anything about beta. It's an invite only service. Beta is not a thing you put in messages.

File: signup/handler/signup.go [hunk 3]
```diff
@@ -292,13 +308,14 @@ func (e *Signup) verify(ctx context.Context, req *signup.VerifyRequest, rsp *sig
...
-		return err
+		logger.Errorf("Error when marking %v verified: %v", req.Email, err)
+		return merrors.InternalServerError("signup.Verify", internalErrorMsg)
 	}
 
 	// At this point the user should be allowed, only making this call to return namespaces
 	namespaces, isAllowed := e.isAllowedToSignup(ctx, req.Email)
 	if !isAllowed {
-		return merrors.Forbidden("signup.notallowed", "user has not been invited to sign up")
+		return merrors.Forbidden("signup.Verify.not_allowed", notInvitedErroMsg)
```
Comment (ben-toogood) [inline signup/handler/signup.go:318]: Can we now use _ here and instead format as `signup.Verify.NotAllowed`?

File: signup/handler/signup.go [hunk 4]
```diff
@@ -337,7 +354,7 @@ func (e *Signup) completeSignup(ctx context.Context, req *signup.CompleteSignupR
 
 	namespaces, isAllowed := e.isAllowedToSignup(ctx, req.Email)
 	if !isAllowed {
-		return merrors.Forbidden("signup.notallowed", "user has not been invited to sign up")
+		return merrors.Forbidden("signup.CompleteSignup.not_allowed", "Email '%v' has not been invited to sign up", req.Email)
```
Comment (ben-toogood) [inline signup/handler/signup.go:357]: Same as above

**Review by ben-toogood — CHANGES_REQUESTED** [review]

File: signup/handler/signup.go [hunk 5]
```diff
@@ -31,6 +31,11 @@ import (
 	mstore "github.com/micro/micro/v3/service/store"
 )
 
+const (
+	internalErrorMsg   = "Something went wrong during signup. Please try again a bit later. If the issue persists, let us know on Slack"
```
Comment (asim) [inline signup/handler/signup.go:35]: "An error occurred during signup. Contact #m3o-support at slack.m3o.com if the issue persists"

File: signup/handler/signup.go [hunk 6]
```diff
@@ -31,6 +31,11 @@ import (
...
+	notInvitedErrorMsg = "We're currently operating an invite only beta and this email address has not been invited"
```
Comment (asim) [inline signup/handler/signup.go:36]: "You have not been invited to the service. Please request an invite on m3o.com"

**Pull request merged**